exec/cnex/array.c

```diff
@@ -27,10 +32,13 @@ Array *array_createArrayFromSize(size_t iElements)
     Array *a = array_createArray();
 
     a->size = iElements;
-    a->data = malloc(sizeof(Cell) * a->size);
+    if (iElements > a->max) {
+        a->max = iElements;
+        a->data = realloc(a->data, sizeof(Cell) * a->max);
```

Comment:
If you're going to have a function to create an array with a specific size, it doesn't make sense to allocate some and then immediately realloc it to what you actually wanted. Just allocate what you want, once.

Reply:
Changed operation of this function to allocate its own `Array`, and allocate (only once) enough room for the required elements.